### PR TITLE
[PROD3-90] Add useSubscriptions and type

### PR DESCRIPTION
**Pull request by letbueno**
# Description

- Add hook of subscriptions and type to return user subscriptions and cancel subscription request

## Does this pull request close any open issues?

- Put here the issue that must be closed with this PR
- eg: closes #prod3-90

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] It is urgent
- [ ] <span style="color:#d44037">Dangerous change</span>

### CODE REVIEW

File: src/hooks/apiHooks/useSubscriptions/index.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  function userSubscriptions(id?: string | number) {
+    const {
+      refetch,
+      isLoading,
+      data: subscriptions,
+    } = useApi<Subscription>({
+      key: "userSubscriptions",
+      fetchMethod: () => {
+        if (!id) return emptyRequest();
+        return subscriptionApi.getUserSubscription(id);
```
Comment (thaisribon) [inline src/hooks/apiHooks/useSubscriptions/index.ts:16]: I think here should be userId, what do you think?

File: src/hooks/apiHooks/useSubscriptions/index.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      },
+    });
+
+    return { subscriptions, refetch, isLoading };
+  }
+  async function cancelSubscription(subscriptionId: string | number) {
+    const { data } = await subscriptionApi.cancelSubscription(subscriptionId);
```
Comment (thaisribon) [inline src/hooks/apiHooks/useSubscriptions/index.ts:23]: and here should be only id

File: src/hooks/apiHooks/useSubscriptions/index.ts (hunk 1 above)
Comment (letbueno) [reply src/hooks/apiHooks/useSubscriptions/index.ts]: yess, i changed

File: src/hooks/apiHooks/useSubscriptions/index.ts (hunk 2 above)
Comment (letbueno) [reply src/hooks/apiHooks/useSubscriptions/index.ts]: i changed :))

**Review by thaisribon — APPROVED** [review]

**Pull request merged**